keps/973-workload-priority/README.md

```diff
@@ -0,0 +1,349 @@
+When using k8s `PriorityClass`, a `priorityClassSource` field has the "scheduling.k8s.io/PriorityClass" value.
+
+```golang
+type WorkloadSpec struct {
+  ...
+  PriorityClassSource string `json:"priorityClassSource,omitempty"`
+  ...
+}
+```
+
+### How to use WorkloadPriorityClass on Job
+
+The `workloadPriorityClass` is specified through a label `kueue.x-k8s.io/priority-class`.
+
+```yaml
+# sample-priority-class.yaml
+apiVersion: kueue.x-k8s.io/v1alpha1
+kind: WorkloadPriorityClass
+metadata:
+  name: sample-priority
+value: 10000
+description: "Sample priority"
+---
+# sample-job.yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: sample-job-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: sample-priority
+spec:
+  parallelism: 3
+  completions: 3
+  suspend: true
+  template:
+    spec:
+      containers:
+      - name: dummy-job
+        image: gcr.io/k8s-staging-perf-tests/sleep:latest
+      restartPolicy: Never
+```
+
+The following workload is generated by the yaml above.
+The `PriorityClass` field can accept both k8s `PriorityClass` and `workloadPriorityClass` names as values.
+To distinguish, when using `workloadPriorityClass`, a `priorityClassSource` field has the "kueue.x-k8s.io/WorkloadPriorityClass" value.
+When using k8s `PriorityClass`, a `priorityClassSource` field has the "scheduling.k8s.io/PriorityClass" value.
+
+```yaml
+# sample-workload.yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: job-sample-job-jf5fb-f5982
+spec:
+  priorityClassSource: kueue.x-k8s.io/WorkloadPriorityClass
+  priorityClassName: sample-priority
+  priority: 10000
+  queueName: user-queue
+  podSets:
+  - count: 3
+    name: dummy-job
+    template:
+      spec:
+        containers:
+        - image: gcr.io/k8s-staging-perf-tests/sleep:latest
+          name: dummy-job
+```
+
+In this example, since the `WorkloadPriorityClassName` of `sample-job` is set to `sample-priority`, the `priority` of the `sample-job` will be set to 10,000.
+During queuing and preemption of the workload, this priority value will be used in the calculations.
+
+### How to use WorkloadPriorityCLass on MPIJob
+
+The `workloadPriorityClass` is specified through a label `kueue.x-k8s.io/priority-class`.
+This is same as other CRDs like `RayJob`.
+
+```yaml
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: pi
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/priority-class: sample-priority
+spec:
+.....
+```
+
+### How workloads are created from Jobs
+
+There are three scenarios for creating a workload from a job.
+The same applies to CRDs other than `Job` (such as `RayJob`).
+
+1. A job specifies both `workload's priority` and `pod's priority`
+2. A job specifies only `workload's priority`
+3. A job specifies only `pod's priority`
```

Comment:
I think there are more cases. Can you add cases if the jobFrameworks, such as `MPIJob`, implement the `JobWithPriorityClass` interface? Which `PriorityClass` is used?

https://github.com/kubernetes-sigs/kueue/blob/9efe0d713b5937ca590efe21bb4ae6f7b15bf196/pkg/controller/jobframework/interface.go#L69-L72

Reply:
I will add cases, thanks.

I thought this `JobWithPriorityClass` interface is for `pod's priority`.
If an MPIJob has both `PriorityClass` and `kueue.x-k8s.io/priority-class`, the `kueue.x-k8s.io/priority-class` is used as 'workload's priority'.